filters/filter.go

```diff
@@ -37,25 +37,26 @@ func HasRequiredRoles(viewRoles []string) beego.FilterFunc {
 			payload,
 			fmt.Sprintf("https://%s/claims/roles", conf.App.Domain),
 		)
-		if tokens.CheckRoles(tokenRoles, viewRoles[0]) {
+		if tokens.CheckRoles(tokenRoles, viewRoles) {
 			return
 		}
-		ctx.Redirect(http.StatusUnauthorized, "/403")
+		ctx.Redirect(http.StatusForbidden, "/403")
 	}
 }
 
 // HasRequiredPermissions checks permissions
-func HasRequiredPermissions(viewPermissions []string) beego.FilterFunc {
+func HasRequiredPermissions(viewPermissions map[string][]string) beego.FilterFunc {
 	return func(ctx *context.Context) {
 		payload, err := reflections.GetField(ctx.Request, "auth_jwt")
 		if err != nil {
 			ctx.Redirect(http.StatusUnauthorized, "/401/Invalid_Authorization_Token'")
 		}
-		tokenPermissions, err := reflections.GetField(
+		val, err := reflections.GetField(
```

Comment:
Lastly, what type would the "val" variable be here? It's meant to be tokenPermissions, is that a string or array of strings?